### ISSUE TITLE
Warn on stdout output in a Debug or Display implementation

### ISSUE THREAD

**Issue by NieDzejkob** [rust-lang/rust-clippy#6705]
### What it does

Detect uses of `print` (instead of `write`) and similar inside of implementations of formatting traits (Debug, Display, LowerHex, etc.)

### Categories (optional)

- Kind: correctness

*What is the advantage of the recommended code over the original code*

Using print will appear to work correctly until one tries to print somewhere other than stdout (a file, string buffer, ...)

### Drawbacks

None.

### Example

```rust
impl fmt::Display for Test {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        println!("Hello");
        writeln!(f, "there!")
    }
}
```

Should be written as:

```rust
impl fmt::Display for Test {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        writeln!(f, "Hello")?;
        writeln!(f, "there!")
    }
}
```

**Comment by unvalley** [comment]
@NieDzejkob (cc: @xFrednet )
This may already be implemented (ref: https://rust-lang.github.io/rust-clippy/master/#print_in_format_impl).
Please tell me if I'm incorrect.

**Comment by xFrednet** [comment]
The lint looked familiar. Guess this was a duplicate issue. Thank you for pointing that out @unvalley! :upside_down_face:
